### PR TITLE
new PI plugin added for TotemDAQMapping class

### PR DESCRIPTION
**Pull request by Kszemka**
#### PR description:

This PR adds TotemDAQMapping class to Payload Inspector module so to be able to visualize it on dedicated PI tab on the cms-conddb website. It consists of a one plot that prints out all the information stored in the payload.

Also - in .clang-format - doubled line cleaned up.
 
#### PR validation:

The code can be tested by a script CondCore/CTPPSPlugins/test/testDAQMapping.sh also created in the PR (it uses getPayloadData.py from CondCore/Utilities) - test checks whether the new plugin is properly added to PI software and therefore it generates proper output file.

#### If this PR is a backport please specify the original PR and why you need to backport that PR. If this PR will be backported please specify to which release cycle the backport is meant for:
No backport.

### CODE REVIEW

File: .clang-format [hunk 1]
```diff
@@ -15,5 +15,4 @@ AllowShortLoopsOnASingleLine: false
 BinPackParameters: false
 AlwaysBreakTemplateDeclarations: Yes
 ReflowComments: false
-BinPackArguments: false
-BinPackParameters: false
+BinPackArguments: false
```
Comment (makortel) [inline .clang-format:18]: The `BinPackArguments` and `BinPackParameters` are two different options (see https://clang.llvm.org/docs/ClangFormatStyleOptions.html#binpackarguments), please undo this change.

Comment (Kszemka) [reply .clang-format:18]: I am aware, please see line 15 and 19 - doubled BinPackParameters.

Comment (makortel) [reply .clang-format:18]: Ah, right, thanks. Could you then do this change in a separate PR?

Comment (Kszemka) [reply .clang-format:18]: Sure, I'll put it in a separate PR :)

File: .clang-format [hunk 2]
```diff
@@ -16,4 +16,4 @@ BinPackParameters: false
 AlwaysBreakTemplateDeclarations: Yes
 ReflowComments: false
 BinPackArguments: false
-BinPackParameters: false
+BinPackParameters: false
```
Comment (mmusich) [inline .clang-format:19]: this file now has a newline.

File: CondCore/CTPPSPlugins/interface/DAQMappingPayloadInspectorHelper.h [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+
+  bool fill() override {
+    auto tag = cond::payloadInspector::PlotBase::getTag<0>();
+    auto tagname = tag.name;
+    auto iov = tag.iovs.back();
+    auto m_payload = this->fetchPayload(std::get<1>(iov));
+
+    if (m_payload != nullptr) {
+      std::stringstream payloadInfo, lineCountStream;
+      //CTPPSDetId(TotemVFATInfo->TotemSymbID->unsignedint).subdetId()
```
Comment (mmusich) [inline CondCore/CTPPSPlugins/interface/DAQMappingPayloadInspectorHelper.h:56]: useful?

File: CondCore/CTPPSPlugins/interface/DAQMappingPayloadInspectorHelper.h [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+      canvas.SaveAs(fileName.c_str());
+
+      return true;
+    } else {
+      return false;
+    }
+  }
+};
+
+#endif
```
Comment (mmusich) [inline CondCore/CTPPSPlugins/interface/DAQMappingPayloadInspectorHelper.h:92]: misses a newline.

File: CondCore/CTPPSPlugins/plugins/DAQMapping_PayloadInspector.cc [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+#include "CondCore/CondDB/interface/Time.h"
+#include "CondCore/CondDB/interface/PayloadReader.h"
+#include "CondFormats/PPSObjects/interface/TotemDAQMapping.h"
+#include "CondCore/CTPPSPlugins/interface/DAQMappingPayloadInspectorHelper.h"
+
+namespace {
+  typedef DAQMappingPayloadInfo<TotemDAQMapping> DAQMappingPayloadInfo_Text;
+}
+
+PAYLOAD_INSPECTOR_MODULE(TotemDAQMapping) { PAYLOAD_INSPECTOR_CLASS(DAQMappingPayloadInfo_Text); }
```
Comment (mmusich) [inline CondCore/CTPPSPlugins/plugins/DAQMapping_PayloadInspector.cc:18]: missing a newline.

File: CondCore/CTPPSPlugins/src/plugin.cc [hunk 6]
```diff
@@ -39,14 +43,16 @@ REGISTER_PLUGIN(CTPPSPixelDAQMappingRcd, CTPPSPixelDAQMapping);
...
 REGISTER_PLUGIN(CTPPSOpticsRcd, LHCOpticalFunctionsSetCollection);
 REGISTER_PLUGIN(PPSDirectSimulationDataRcd, PPSDirectSimulationData);
 REGISTER_PLUGIN(PPSPixelTopologyRcd, PPSPixelTopology);
 REGISTER_PLUGIN(PPSAlignmentConfigRcd, PPSAlignmentConfig);
 REGISTER_PLUGIN(PPSAlignmentConfigurationRcd, PPSAlignmentConfiguration);
+REGISTER_PLUGIN(TotemAnalysisMaskRcd, TotemAnalysisMask);
+REGISTER_PLUGIN(TotemReadoutRcd, TotemDAQMapping);
 
-REGISTER_PLUGIN_INIT(PPSAssociationCutsRcd, PPSAssociationCuts, InitAssociationCuts);
+REGISTER_PLUGIN_INIT(PPSAssociationCutsRcd, PPSAssociationCuts, InitAssociationCuts);
```
Comment (mmusich) [inline CondCore/CTPPSPlugins/src/plugin.cc:58]: misses a newline.

File: CondCore/CTPPSPlugins/test/testDAQMapping.sh [hunk 7]
```diff
@@ -0,0 +1,29 @@
...
+if [ "$1" == "run" ]
+then
+    mkdir -p CondCore/CTPPSPlugins/test/results
+    if [ -f *.png ]; then    
+    rm *.png
+    fi
+
+    echo "Testing DAQMapping info"
+
+    python3 CondCore/Utilities/scripts/getPayloadData.py \
```
Comment (mmusich) [inline CondCore/CTPPSPlugins/test/testDAQMapping.sh:12]:
```suggestion
    getPayloadData.py \
```

this presumes that the one has checked out ` CondCore/Utilities` in the first place.
`getPayloadData.py ` is a script avalailble in the path.

File: CondCore/CTPPSPlugins/test/testDAQMapping.sh [hunk 8]
```diff
@@ -0,0 +1,29 @@
...
+        --test    
+
+    mv *.png CondCore/CTPPSPlugins/test/results/DAQMapping_TextInfo.png 
+     
+elif [ "$1" == "clear" ]
+then
+    rm -rf CondCore/CTPPSPlugins/test/results
+
+else 
+    echo "Wrong option"
```
Comment (mmusich) [inline CondCore/CTPPSPlugins/test/testDAQMapping.sh:28]: would be nice to prompt the user what are the available options....

**Review by mmusich — COMMENTED** [review]
some unrequested feedback by a non-reviewer ;)

File: CondCore/CTPPSPlugins/test/testDAQMapping.sh [hunk 9]
```diff
@@ -0,0 +1,29 @@
...
+        --time_type Run \
+        --iovs '{"start_iov": "283820", "end_iov": "283820"}' \
+        --db Prod \
+        --test    
+
+    mv *.png CondCore/CTPPSPlugins/test/results/DAQMapping_TextInfo.png 
+     
+elif [ "$1" == "clear" ]
+then
+    rm -rf CondCore/CTPPSPlugins/test/results
```
Comment (mmusich) [inline CondCore/CTPPSPlugins/test/testDAQMapping.sh:25]:
```suggestion
    rm -rf $CMSSW_BASE/src/CondCore/CTPPSPlugins/test/results
```

**Review by mmusich — COMMENTED** [review]
also if you apply these suggestions, the output folder doesn't actually depend from where the script is executed.

File: .clang-format (hunk 2 above)
Comment (rsreds) [reply .clang-format:19]: Quick question, as I'm still learning, aren't missing newlines caught by code checks?

Comment (makortel) [reply .clang-format:19]: Code checks (that run `clang-format` and `clang-tidy`) inspect only C++ code.

Comment (rsreds) [reply .clang-format:19]: Ah, got it. Thanks.

File: CondCore/CTPPSPlugins/test/testDAQMapping.sh (hunk 7 above)
Comment (mmusich) [reply CondCore/CTPPSPlugins/test/testDAQMapping.sh:12]: @Kszemka looks like this suggestion was not applied. Any reason for that?

Comment (Kszemka) [reply CondCore/CTPPSPlugins/test/testDAQMapping.sh:12]: accidentally missed, I already updated. c:

**Comment by grzanka** [comment]
@makortel is there anything else we should do in this PR ?

**Comment by makortel** [comment]
> @makortel is there anything else we should do in this PR ?

Question should really be for @cms-sw/db-l2

**Comment by perrotta** [comment]
please test

**Comment by perrotta** [comment]
+db

**Comment by rappoccio** [comment]
+1
